### DISCUSSION TITLE
How to focus the closest prev space after destroying a space?

### DISCUSSION THREAD

**Discussion by leandwo** [asmvik/yabai#964] [Q&A] [answered]
When I run `yabai -m space --destroy` yabai focuses the always very first space which doesn't feel natural to me. How can I get yabai to focus the prev space instead?

**Comment by leandwo** [comment, marked as answer]
I found a workaround but it feels jank beacuase the first screen flashes:
```zsh
hyper - x:                                                                                                                                                                          
  current_space=$(yabai -m query --spaces --space | jq -r '.index'); \                                                                                                              
  yabai -m space --destroy; \                                                                                                                                                       
  yabai -m space --focus $((current_space - 1)) 
```

**Reply by asmvik** [reply]
You should just perform the operation in the reverse order. First focus the prev space and if successful, then destroy the space you came from; something like: `hyper - x : yabai -m space --focus prev && yabai -m space recent --destroy`

**Reply by leandwo** [reply]
Ah, that works out way better. Thank you!

**Reply by aurexav** [reply]
Expect:
If my previous work space is 1, current work space is 3, this will lead me to the space 2.

Actual:
This will kill the space 2 and lead me to the space 1. I don't know why.

@koekeishiya

**Reply by aurexav** [reply]
```
yabai -m space --focus recent && sleep 1 && yabai -m space recent --destroy
```

This works for me. It seems the status has not changed if too fast.

**Reply by metafates** [reply]
@AurevoirXavier Thanks! I had the same issue. By the way, I would recommend using `sleep 0.01` instead of 1, still enough for yabai to update

**Comment by Guth-Alak** [comment]
just another option, this one goes to the last workspace instead of the first one

`yabai -m space --destroy ;\ yabai -m space --focus last`

**Reply by Guth-Alak** [reply]
although that does still flash the space1 briefly, this order avoids that 

`yabai -m space --focus last && yabai -m space recent --destroy || yabai -m space --focus prev && yabai -m space next --destroy`
